src/hotspot/share/opto/mulnode.cpp

```diff
@@ -616,6 +616,16 @@ const Type* MulHiValue(const Type *t1, const Type *t2, const Type *bot) {
       (t1 == Type::BOTTOM) || (t2 == Type::BOTTOM) )
     return bot;
 
+  // If the both inputs are constants
+  const TypeLong *longType1 = t1->is_long();
+  const TypeLong *longType2 = t2->is_long();
+  if(longType1 && longType2 && longType1->is_con() && longType2->is_con()){
```

Comment:
```suggestion
  if(longType1 != nullptr && longType2 != nullptr && longType1->is_con() && longType2->is_con()){
```
I know, it seems a bit fussy, but that's the way we do it.